### PR TITLE
【Hackathon 7th No.20】为 Paddle 新增 Tensor.resize_ -part

### PR DESCRIPTION
**Pull request by NKNaN**
### PR Category

User Experience

### PR Types

New features

### Description

- rfc: https://github.com/PaddlePaddle/community/pull/968
- 中文文档：https://github.com/PaddlePaddle/docs/pull/7026

### CODE REVIEW

**Comment by luotao1** [comment]
test_inplace (Failed)

File: test/legacy_test/test_inplace.py [hunk 1]
```diff
@@ -2283,5 +2283,162 @@ def test_inplace_api(self):
...
+
+            self.assertRaises(ValueError, leaf_inplace_error)
+
+
+@unittest.skipIf(
+    not paddle.base.core.is_compiled_with_cuda()
+    or not paddle.base.core.is_float16_supported(paddle.CUDAPlace(0)),
+    "core is not compiled with CUDA and not support the float16",
+)
+class TestDygraphInplaceResizeP16(TestDygraphInplaceResize):
```
Comment (zxcd) [inline test/legacy_test/test_inplace.py]: FP16?

File: python/paddle/tensor/creation.py [hunk 2]
```diff
@@ -3379,3 +3379,75 @@ def set_(
...
+    Resize ``x`` with specified ``shape``.
+
+    Args:
+        x (Tensor): An arbitrary Tensor. The data type supports ``bfloat16``, ``float16``, ``float32``, ``float64``,
+            ``bool``, ``int8``, ``int16``, ``int32``, ``int64``, ``uint8``, ``complex64`` or ``complex128``.
+        shape (list|tuple, optional): Define the target shape. Each element of it should be integer.
+        fill_zero (bool, optional): If the size of specified ``shape`` is greater than the original Tensor size, the
+            new Tensor will be filled with zero if ``fill_zero`` is True. Default: False, which means the filled value
+            will be undetermined.
+        name (str|None, optional): Name for the operation (optional, default is None). For more information, please refer to :ref:`api_guide_Name`.
```
Comment (zxcd) [inline python/paddle/tensor/creation.py:3411]: Where did this parameter come from?

File: python/paddle/tensor/creation.py [hunk 3]
```diff
@@ -3379,3 +3379,75 @@ def set_(
...
+                'int8',
+                'int16',
+                'int32',
+                'int64',
+                'uint8',
+                'complex64',
+                'complex128',
+            ],
+            'resize',
+        )
```
Comment (zxcd) [inline python/paddle/tensor/creation.py:3455]: miss data format check for `shape`

File: python/paddle/tensor/creation.py (hunk 2 above)
Comment (NKNaN) [reply python/paddle/tensor/creation.py:3411]: To make the functionality aligned with numpy (when calling `ndarray.resize()` )

image

File: test/legacy_test/test_inplace.py (hunk 1 above)
Comment (NKNaN) [reply test/legacy_test/test_inplace.py]: yes, revised.

File: python/paddle/tensor/creation.py (hunk 3 above)
Comment (NKNaN) [reply python/paddle/tensor/creation.py:3455]: added

File: python/paddle/tensor/creation.py (hunk 2 above)
Comment (zxcd) [reply python/paddle/tensor/creation.py:3411]: Then you should add this parameter to interface?

Comment (NKNaN) [reply python/paddle/tensor/creation.py:3411]: > Then you should add this parameter to interface?

Do you mean I should not add this parameter and functionality to the interface?

Comment (zxcd) [reply python/paddle/tensor/creation.py:3411]: your interface have three inputs, but doc get four?

Comment (NKNaN) [reply python/paddle/tensor/creation.py:3411]: oh I forgot to include the parameter `name`. 
added.

File: python/paddle/tensor/creation.py (hunk 3 above)
Comment (zxcd) [reply python/paddle/tensor/creation.py:3455]: also converge this codes in unit test?

Comment (NKNaN) [reply python/paddle/tensor/creation.py:3455]: image
covered here for `shape` data format check.

**Review by zxcd — APPROVED** [review]
LGTM

**Review by jeff41404 — APPROVED** [review]
LGTM

**Review by sunzhongkai588 — APPROVED** [review]
LGTM for docs

**Pull request merged**